app/transactions/ams/steps/create_aapb_admin_data.rb

```diff
@@ -51,22 +51,12 @@ def save_aapb_admin_data(change_set)
       end
 
       def set_admin_data_attributes(admin_data, change_set)
-        AdminData.attributes_for_update.each do |k|
-          # Some attributes are serialized on AdminData, so always send an array
-          k_string = k.to_s
-          if should_empty_admin_data_value?(k, change_set)
-            AdminData::SERIALIZED_FIELDS.include?(k) ? admin_data.send("#{k}=", Array.new) : admin_data.send("#{k}=", nil)
-          elsif change_set.fields[k_string].present?
-            AdminData::SERIALIZED_FIELDS.include?(k) ? admin_data.send("#{k}=", Array(change_set.fields[k_string])) : admin_data.send("#{k}=", change_set.fields[k_string].to_s)
-          end
+        admin_data_values = change_set.fields.slice(*AdminData.attributes_for_update).compact
```

Comment:
`#slice` only works if the keys are the same type. We know `AdminData.attributes_for_update` are strings, is `change_set.fields`?

Reply:
Yeah, pretty sure from debugging the specs. Also the previous comparison assumed string keys: `change_set.fields[k_string].present?`